tests/pipeline/test_factor.py

```diff
@@ -585,6 +586,34 @@ def test_returns(self, seed_value, window_length):
 
         check_allclose(expected, out)
 
+    @parameterized.expand([
+        (100, 15),
+        (101, 4),
+        (102, 100),
+        ])
+    def test_percentchange(self, seed_value, window_length):
+
+        pct_change = PercentChange(
+            inputs=(), 
+            window_length=window_length
+        )
+
+        today = datetime64(1, 'ns')
+        assets = arange(7)
+
+        seed(seed_value)  # Seed so we get deterministic results.
+        test_data = randn(window_length, 7)
+        test_data[0] = array([1, 2, 1, -1, -1, 0, nan])
+        test_data[-1] = array([2, 1, -2, 2, -2, 1, 1])
+
+        # Calculate the expected percent change
+        expected = array([1, -0.5, -3, 3, -1, inf, nan])
+
+        out = empty((7,), dtype=float)
+        pct_change.compute(today, assets, out, test_data)
+
+        check_allclose(expected, out)
+
```

Comment:
Another case you could test for is passing a `window_length` of 1 and making sure an exception is raised. This can be done with something like:

```python
with self.assertRaises(ValueError):
    PercentChange(inputs=(), window_length=1)
```